src/index/IndexImpl.Text.cpp

```diff
@@ -150,25 +138,30 @@ void IndexImpl::buildTextIndexFile(
   vocab_ = RdfsVocabulary{};
   readConfiguration();
   {
-    auto [b, k] = bAndKForBM25;
-    storeTextScoringParamsInConfiguration(textScoringMetric, b, k);
+    auto [b, k] = config.bAndKForBM25_;
+    storeTextScoringParamsInConfiguration(config.textScoringMetric_, b, k);
   }
   vocab_.readFromFile(onDiskBase_ + VOCAB_SUFFIX);
 
   scoreData_ = {vocab_.getLocaleManager(), textScoringMetric_,
                 bAndKParamForTextScoring_};
 
   // Build the text vocabulary (first scan over the text records).
-  processWordsForVocabulary(wordsFile, addWordsFromLiterals);
+  processWordsForVocabulary(
+      config.useDocsFileForVocabulary_ ? docsFile : wordsFile,
+      config.addWordsFromLiterals_, config.useDocsFileForVocabulary_);
```

Comment:
Can you pass in the complete config into this function, you need almost all members
(maybe then the `docsFile` and `wordsFile` variables here become unused.